util/pace/util/grid/stretch_transformation.py

```diff
@@ -1,19 +1,24 @@
 import copy
-from typing import Tuple, Union
 
 import numpy as np
 
+
+try:
+    import cupy as cp
+except ImportError:
+    cp = None
 from pace.util import Quantity
 
 
 def direct_transform(
     *,
-    lon: Union[Quantity, np.ndarray],
-    lat: Union[Quantity, np.ndarray],
+    lon,
+    lat,
     stretch_factor: np.float_,
     lon_target: np.float_,
     lat_target: np.float_,
-) -> Tuple[Union[np.ndarray, Quantity], Union[np.ndarray, Quantity]]:
+    numpy_module,
```

Comment:
nit: Please call this argument `np` as we do in the grid generation (e.g. util/pace/util/grid/gnomonic.py) in this and the other functions.